File: MCP服务器/browser-automation/lightpanda-io_gomcp_73e4ecb_go/mcp/tool.go
// Copyright 2025 Lightpanda (Selecy SAS)
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package mcp

type Schema any

type SchemaType struct {
	Type        string `json:"type"`
	Description string `json:"description"`
}

type schemaString SchemaType

func NewSchemaString(description string) schemaString {
	return schemaString(SchemaType{Type: "string", Description: description})
}

type Properties map[string]Schema

type schemaObject struct {
	SchemaType
	Properties           Properties `json:"properties"`
	AdditionalProperties bool       `json:"additionalProperties"`
}

func NewSchemaObject(p map[string]Schema) schemaObject {
	return schemaObject{
		SchemaType: SchemaType{Type: "object"},
		Properties: p,
	}
}

type Tool struct {
	Name        string       `json:"name"`
	Description string       `json:"description,omitempty"`
	InputSchema schemaObject `json:"inputSchema"`
	// TODO annotations
}

File: reference-mcp-servers/ai/eagurin_myaiserv_f917574_python/app/models/mcp/tool.py
"""
Модель инструмента MCP.
"""

from typing import Any, Dict

from pydantic import BaseModel, Field


class Tool(BaseModel):
    """
    Модель инструмента MCP.

    Examples:
        >>> tool = Tool(
        ...     name="calculate_sum",
        ...     description="Add two numbers",
        ...     input_schema={
        ...         "type": "object",
        ...         "properties": {
        ...             "a": {"type": "number"},
        ...             "b": {"type": "number"}
        ...         },
        ...         "required": ["a", "b"]
        ...     }
        ... )

    Attributes:
        name: Уникальное имя инструмента
        description: Описание функциональности
        input_schema: JSON Schema входных параметров
        requires_approval: Требуется ли одобрение
    """

    name: str = Field(..., description="Уникальное имя инструмента")
    description: str = Field(..., description="Описание функциональности")
    input_schema: Dict[str, Any] = Field(
        ..., description="JSON Schema входных параметров"
    )
    requires_approval: bool = Field(False, description="Требуется ли одобрение")
